File: [23.12]Design-Pattern/Java/src/behavioral/state/Main.java
package behavioral.state;

public class Main {
    public static void main(String[] args) {
        Laptop laptop = new Laptop();

        System.out.println("[초기 상태]");
        laptop.currentStatePrint();
        laptop.typeButtonPush();

        System.out.println();
        System.out.println("[전원 버튼 클릭]");
        laptop.powerButtonPush();
        laptop.currentStatePrint();
        laptop.typeButtonPush();

        System.out.println();
        System.out.println("[전원 버튼 클릭]");
        laptop.powerButtonPush();
        laptop.currentStatePrint();
        laptop.typeButtonPush();

    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/state/Main.kt
package behavioral.state

fun main() {
    val laptop = Laptop()

    println("[초기 상태]")
    laptop.currentState()
    laptop.typeButtonPush()

    println()
    println("[전원 버튼 클릭]")
    laptop.powerButtonPush()
    laptop.currentState()
    laptop.typeButtonPush()

    println()
    println("[전원 버튼 클릭]")
    laptop.powerButtonPush()
    laptop.currentState()
    laptop.typeButtonPush()

}